presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -210,13 +242,80 @@ public void addInput(Page page)
 
     private void handleTooLargePredicate()
     {
-        // The resulting predicate is too large, allow all probe-side values to be read.
-        dynamicPredicateConsumer.accept(TupleDomain.all());
+        // The resulting predicate is too large
+        if (minMaxChannels.isEmpty()) {
+            // allow all probe-side values to be read.
+            dynamicPredicateConsumer.accept(TupleDomain.all());
+        }
+        else {
+            // convert to min/max per column for orderable types
+            for (Integer channelIndex : minMaxChannels) {
+                Block block = blockBuilders[channelIndex].build();
+                updateMinMaxValues(block, channelIndex);
+            }
+        }
         // Drop references to collected values.
         valueSets = null;
         blockBuilders = null;
     }
 
+    private void updateMinMaxValues(Block block, int channelIndex)
+    {
+        if (minValues == null || maxValues == null) {
+            return;
+        }
+        Type type = channels.get(channelIndex).type;
+        int minValuePosition = -1;
+        int maxValuePosition = -1;
+
+        for (int position = 0; position < block.getPositionCount(); ++position) {
+            if (block.isNull(position)) {
+                continue;
+            }
+            if (minValuePosition == -1) {
+                // First non-null value
+                minValuePosition = position;
+                maxValuePosition = position;
```

Comment:
use `continue`. Then you can remove `else`

Reply:
done